tests/conftest.py

```diff
@@ -70,36 +27,36 @@ def ca_ssl_context(cert_authority: trustme.CA) -> ssl.SSLContext:
     return ctx
 
 
-@pytest.fixture(scope="session")
-def example_org_cert(cert_authority: trustme.CA) -> trustme.LeafCert:
-    return cert_authority.issue_cert("example.org")
+def wait_until_pproxy_serve_on_port(host: str, port: int):
+    while True:
+        try:
+            sock = socket.create_connection((host, port))
+        except ConnectionRefusedError:
+            time.sleep(0.25)
+        else:
+            sock.close()
+            break
 
 
 @pytest.fixture(scope="session")
-def example_org_cert_path(example_org_cert: trustme.LeafCert) -> typing.Iterator[str]:
-    with example_org_cert.private_key_and_cert_chain_pem.tempfile() as tmp:
-        yield tmp
+def proxy_server() -> typing.Iterator[URL]:
+    http_proxy_host = "127.0.0.1"
+    http_proxy_port = 8080
 
+    proc = None
+    try:
+        command_str = f"pproxy -l http://{http_proxy_host}:{http_proxy_port}/"
+        command = shlex.split(command_str)
```

Comment:
Don't think we need `shlex` here, if we build the list of arguments directly…?

```suggestion
        command = ["pproxy", "-l", f"http://{http_proxy_host}:{http_proxy_port}/"]
```